### PR TITLE
Creating announcements, instead of message popup.

### PR DESCRIPTION
**Pull request by kimhansen**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] If this change fixes a bug (or a performance problem), is a regression
  test (or a benchmark) included?
- [ ] Is a documentation update included (if this change modifies
  existing APIs, or introduces new ones)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### CODE REVIEW

File: web/se/static/components/se/shareregistry/shareregistry_announcement_details.tag [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            <input show={ state != 'details' } type="submit" class="btn waves-effect" value="Send"></input>
+        </div>
+    </form>
+
+    <script>
+        let thizz = this
+        thizz.announcement = opts.announcement
+        thizz.state = opts.state
+        thizz.title = "Announcement Details"
+        thizz.canEditShareregistry = eval(riot.app.store.getCanEditShareregistry())
```
Comment (rdmrocha) [inline web/se/static/components/se/shareregistry/shareregistry_announcement_details.tag]: why the eval? getCanEditShareregistry already returns a boolean so I don't really understand the need for this

File: web/se/static/components/se/shareregistry/shareregistry_announcement_details.tag [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            else {
+                thizz.refs.subject.value = ""
+                thizz.refs.body.value = ""
+            }
+            thizz.update()
+        }
+
+        thizz.submitCancel = function() {
+            if (thizz.state == 'add' || thizz.state == 'details') {
+                riot.app.events.triggerOrganisationChange(thizz.announcement.organisationId);
```
Comment (rdmrocha) [inline web/se/static/components/se/shareregistry/shareregistry_announcement_details.tag]: why are we triggering an organisationChange on cancel?? are you trying to force an update on the other views?

File: web/se/static/components/se/shareregistry/shareregistry_announcements.tag [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    <div class="empty" if={ state == "" } show={ announcements.length <= 0 }>
+        List is empty
+    </div>
+
+    <script>
+        let thizz = this;
+        thizz.announcements = [];
+        thizz.announcement = null;
+        thizz.canEditShareregistry = eval(riot.app.store.getCanEditShareregistry())
```
Comment (rdmrocha) [inline web/se/static/components/se/shareregistry/shareregistry_announcements.tag]: same as above: why eval?

File: web/se/static/components/se/shareregistry/shareregistry_announcements.tag [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+        // called when receiving data from server
+        thizz.onOrganisationDataReceived = function(data) {
+            thizz.announcements = data.shareholderMessage
```
Comment (rdmrocha) [inline web/se/static/components/se/shareregistry/shareregistry_announcements.tag]: data can contain errors and we should check for them

Comment (rdmrocha) [reply web/se/static/components/se/shareregistry/shareregistry_announcements.tag]: I would do something like **_thizz. announcements = formatAnnouncements(data. shareholderMessage)_** in order to keep formatAnnouncements generic, reusable and not dependent on local context variables. There's nothing wrong with this approach in terms of functionality, but it's a best practice to only use global or context vars when needed.

File: web/se/static/components/se/shareregistry/shareregistry_announcement_details.tag (hunk 1 above)
Comment (kimhansen) [reply web/se/static/components/se/shareregistry/shareregistry_announcement_details.tag]: I will fix the eval in another PR, as it is not only for announcements.

File: web/se/static/components/se/shareregistry/shareregistry_announcement_details.tag (hunk 2 above)
Comment (kimhansen) [reply web/se/static/components/se/shareregistry/shareregistry_announcement_details.tag]: Yes, removed. Shouldn't be needed.

File: web/se/static/components/se/shareregistry/shareregistry_announcements.tag (hunk 4 above)
Comment (kimhansen) [reply web/se/static/components/se/shareregistry/shareregistry_announcements.tag]: Yes, I have improved this.

**Pull request merged**